src/background/mod.rs

```diff
@@ -41,18 +89,242 @@ pub struct PlanetComponent {
     pub rotation_speed: f32,
 }
 
+/// Component to tag star point light
+#[derive(Component)]
+pub struct StarLightComponent;
+
 /// Rotate planets about their z axis
-pub fn rotate_planet_system(mut query: Query<(&mut Transform, &PlanetComponent)>) {
+pub fn rotate_planet_system(mut query: Query<(&mut Transform, &PlanetComponent)>, time: Res<Time>) {
     for (mut transform, planet) in query.iter_mut() {
-        transform.rotation *= Quat::from_rotation_y(planet.rotation_speed);
+        transform.rotation *= Quat::from_rotation_y(planet.rotation_speed * time.delta_seconds());
+    }
+}
+
+/// Execute the exploding star effect if the game is lost through defense being destroyed
+pub fn on_defeat_star_explode_system(
+    mut run_end_event_reader: EventReader<RunEndEvent>,
+    mut point_light_query: Query<&mut PointLight, With<StarLightComponent>>,
+    mut star_explode_res: ResMut<StarExplodeResource>,
+    time: Res<Time>,
+    backgrounds_res: Res<BackgroundsResource>,
+) {
+    // Check for loss condition from defense objective
+    for event in run_end_event_reader.iter() {
+        if let RunOutcomeType::Defeat(RunDefeatType::DefenseDestroyed) = event.outcome {
+            star_explode_res.started = true;
+        }
     }
+
+    // Update star point light intensity if star explosion active
+    if star_explode_res.started {
+        for mut point_light in point_light_query.iter_mut() {
+            point_light.intensity *= backgrounds_res.star_explode_intensity * time.delta_seconds();
+        }
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum OurGetRandomAssetError {
```

Comment:
I think only the plugin should need to be publicly exposed from this module. 
```suggestion
enum OurGetRandomAssetError {
```